src/System.Security.Cryptography.Cng/tests/SymmetricCngTestHelpers.cs

```diff
@@ -271,6 +273,22 @@ internal static bool SupportsPersistedSymmetricKeys
             }
         }
 
+        private static bool? s_isAdministrator;
+
+        internal static bool IsAdministrator
+        {
+            get
+            {
+                if (!s_isAdministrator.HasValue)
+                {
+                    s_isAdministrator =
+                        new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator);
+                }
```

Comment:
This isn't thread-safe; is this accessed from multiple tests that might run concurrently?